src/Network/Wai/Middleware/Auth.hs

```diff
@@ -297,3 +299,11 @@ getAuthUser = Vault.lookup userKey . vault
 waiMiddlewareAuthVersion :: Version
 waiMiddlewareAuthVersion = Paths.version
 
+-- | Get a response header to delete the users current session.
+--
+-- @since 0.2.0
+getDeleteSessionHeader :: AuthSettings -> IO Header
+getDeleteSessionHeader AuthSettings {..} = do
+  secretKey <- asGetKey
+  -- Delete the cookie by setting an expiry date far in the past.
+  saveCookieValue secretKey asStateKey (-365 * 24 * 3600) ("" :: S.ByteString)
```

Comment:
Setting age in the past looks hacky, plus with `saveCookieValue` the value, even if empty, is being encrypted, so in the end cookie will contains some value that decodes into empty bytestring, which is not what we want.

I'd rather use a helper function like this:
```haskell

deleteCookieValue
  :: Binary value
  => IO Header
deleteCookieValue name = do
  return
    ( "Set-Cookie"
    , toByteString $
      renderSetCookie
        def
          { setCookieName = name
          , setCookieValue = ""
          , setCookiePath = Just "/"
          , setCookieHttpOnly = True
          , setCookieExpires = Just $ UTCTime (fromGregorian 1970 01 01) 0
          })
```

Reply:
That seems much nicer. Pushed a change to implement this change!

Reply:
I am sorry, it was pretty late for me last night too. Could you get rid of the `IO` in `deleteCookieValue` and in `getDeleteSessionHeader`, those are all pure computations. Other than that looks great and almost ready to be merged

Reply:
Oops, completely missed that too! Fixed it.

Where are you based?

Reply:
Minsk, Belarus.

Reply:
Cool! Working from Cambridge, UK myself. Nice to meet you!

Reply:
It is my pleasure :)
I was in your neck of the woods just a couple of days ago, at Haskell Exchange in London.

Reply:
Yeah, I just saw the videos for that they posted. Looks like there were many great talks. I'm considering going next year.